### PR TITLE
fix v2.0.0 metrics bug

### PR DESCRIPTION
**Pull request by MegaByte875**
The metrics have changed to "value=286 name=heartbeat_latency_us.avg.5" format

### CODE REVIEW

File: exporter/exporter.go [hunk 1]
```diff
@@ -489,3 +503,32 @@ func (exporter *NebulaExporter) buildMetricMap(
...
+		}
+		ok, value := getRValue(metrics[t])
+		if !ok {
+			continue
+		}
+		ok, metric := getRValue(metrics[t+1])
+		if !ok {
+			continue
+		}
+		if len(value) > 0 && len(metric) > 0 {
```
Comment (veezhang) [inline exporter/exporter.go:521]: if value != "" && metric != "" { // golangci-lint

File: exporter/exporter.go [hunk 2]
```diff
@@ -489,3 +503,32 @@ func (exporter *NebulaExporter) buildMetricMap(
 	}
 	return exporter
 }
+
+func convertToMap(metrics []string) map[string]string {
+	matches := make(map[string]string)
+	for t := 0; ; t += 2 {
+		if t == len(metrics) {
```
Comment (veezhang) [inline exporter/exporter.go:510]: if len(metrics) is Odd number,  metrics[t+1] will panic. 
`t + 1 >= len(metrics)` will be more safty.

File: exporter/exporter.go [hunk 3]
```diff
@@ -72,6 +72,9 @@ func getNebulaMetrics(ipAddress string, port int32) ([]string, error) {
 
 	metricStr := string(bytes)
 	metrics := strings.Split(metricStr, "\n")
+	if len(metrics) > 0 {
```
Comment (veezhang) [inline exporter/exporter.go:75]: remove the blank line? What if not?

File: exporter/exporter.go [hunk 4]
```diff
@@ -324,23 +327,34 @@ func (exporter *NebulaExporter) CollectMetrics(
 	namespace string,
 	metrics []string,
 	ch chan<- prometheus.Metric) {
-	for _, singleMetric := range metrics {
-		seps := strings.Split(singleMetric, "=")
-		if len(seps) != 2 {
-			continue
-		}
-
+	// status metric
```
Comment (veezhang) [inline exporter/exporter.go:330]: if len(metrics) == 0  return earlier

File: exporter/exporter.go [hunk 5]
```diff
@@ -324,23 +327,34 @@ func (exporter *NebulaExporter) CollectMetrics(
...
-		if err != nil {
-			continue
+		if err == nil {
+			if metric, ok := exporter.metricMap[seps[0]]; ok {
+				ch <- prometheus.MustNewConstMetric(metric.desc, prometheus.GaugeValue, values,
+					name,
+					namespace,
+					componentType,
+				)
+			}
```
Comment (veezhang) [inline exporter/exporter.go:341]: return it, and no need the else any more?

File: exporter/exporter.go [hunk 6]
```diff
@@ -324,23 +327,34 @@ func (exporter *NebulaExporter) CollectMetrics(
...
-			ch <- prometheus.MustNewConstMetric(metric.desc, prometheus.GaugeValue, values,
-				name,
-				namespace,
-				componentType,
-			)
+	} else {
+		// query metrics
+		matches := convertToMap(metrics)
+		for metric, value := range matches {
+			values, err := strconv.ParseFloat(value, 64)
```
Comment (veezhang) [inline exporter/exporter.go:347]: values feel like an array or slice, use value or v ?

Comment (MegaByte875) [reply exporter/exporter.go:347]: right, variable name "v" is better

File: exporter/exporter.go (hunk 2 above)
Comment (MegaByte875) [reply exporter/exporter.go:510]: len(metrics) is even number， but t+1 may  avoid panic and more safety

File: exporter/exporter.go [hunk 7]
```diff
@@ -64,6 +64,7 @@ func getNebulaMetrics(ipAddress string, port int32) ([]string, error) {
 	if err != nil {
 		return []string{}, err
 	}
+	defer resp.Body.Close()
 
 	bytes, err := ioutil.ReadAll(resp.Body)
```
Comment (veezhang) [inline exporter/exporter.go:69]: is it possible '\n' or '\r\n' at the end line?
is it possible non-printable in each line?
is it necessary use buff for http body?
```
	metrics := make([]string, 0)
	scanner := bufio.NewScanner(resp.Body)
	for scanner.Scan() {
		if line := strings.TrimSpace(sc.Text()); line != ""{
			metrics = append(metrics, line)
		}
	}
```

Comment (MegaByte875) [reply exporter/exporter.go:69]: each line end with "\n" and is printable metrics，if  metrics are large use buff more better.

**Review by veezhang — APPROVED** [review]

**Comment by veezhang** [comment]
@MegaByte875 LGTM

File: exporter/exporter.go [hunk 8]
```diff
@@ -324,19 +332,35 @@ func (exporter *NebulaExporter) CollectMetrics(
...
+					name,
+					namespace,
+					componentType,
+				)
+			}
+		}
+		return
+	}
+
+	// query metrics
```
Comment (nianiaJR) [inline exporter/exporter.go:355]: remove the unnecessary comment

**Pull request merged**